nanoFramework.System.Threading/Threading/Tasks/Task.cs

```diff
@@ -260,9 +306,14 @@ protected internal void CompleteWithException(Exception e)
         }
 
         // TODO check .NET
+        /// <summary>
+        /// Get the result of the async operation. Block the calling thread until task completes
+        /// </summary>
         public void GetResult()
         {
-            while (!IsCompleted) ;
+            //while (!IsCompleted) ;
+            resultWaitHandle.WaitOne();
+            resultWaitHandle.Set(); //use it as one shot
             if (Exception != null)
```

Comment:
Please add open and closing brackets, despite being a single line. That's the code style the project is following.